when running test case on config, it may FAIL sometimes

<!--
Please answer these questions before submitting your issue. Thanks!
For questions please use one of our forums: https://go-kratos.dev/docs/getting-started/faq
-->
#### What happened:

when run test case in pkg config, it may FAIL sometimes.

```
➜  config git:(main) go test -v -count=1 ./...
=== RUN   TestConfig
--- PASS: TestConfig (0.00s)
=== RUN   TestDefaultResolver
=== RUN   TestDefaultResolver/test_not_exist_int_env_with_default
=== RUN   TestDefaultResolver/test_string_with_default
=== RUN   TestDefaultResolver/test_int_with_default
=== RUN   TestDefaultResolver/test_bool_with_default
=== RUN   TestDefaultResolver/test_float_without_default
=== RUN   TestDefaultResolver/test_empty_value_with_default
=== RUN   TestDefaultResolver/test_array
=== RUN   TestDefaultResolver/test_${test.value}
=== RUN   TestDefaultResolver/test_$value
=== RUN   TestDefaultResolver/test_$value:default
--- PASS: TestDefaultResolver (0.00s)
    --- PASS: TestDefaultResolver/test_not_exist_int_env_with_default (0.00s)
    --- PASS: TestDefaultResolver/test_string_with_default (0.00s)
    --- PASS: TestDefaultResolver/test_int_with_default (0.00s)
    --- PASS: TestDefaultResolver/test_bool_with_default (0.00s)
    --- PASS: TestDefaultResolver/test_float_without_default (0.00s)
    --- PASS: TestDefaultResolver/test_empty_value_with_default (0.00s)
    --- PASS: TestDefaultResolver/test_array (0.00s)
    --- PASS: TestDefaultResolver/test_${test.value} (0.00s)
    --- PASS: TestDefaultResolver/test_$value (0.00s)
    --- PASS: TestDefaultResolver/test_$value:default (0.00s)
=== RUN   TestDefaultDecoder
--- PASS: TestDefaultDecoder (0.00s)
=== RUN   TestReader_Merge
--- PASS: TestReader_Merge (0.00s)
=== RUN   TestReader_Value
=== RUN   TestReader_Value/json_value
=== RUN   TestReader_Value/yaml_value
--- PASS: TestReader_Value (0.00s)
    --- PASS: TestReader_Value/json_value (0.00s)
    --- PASS: TestReader_Value/yaml_value (0.00s)
=== RUN   TestReader_Source
--- PASS: TestReader_Source (0.00s)
=== RUN   Test_atomicValue_Bool
--- PASS: Test_atomicValue_Bool (0.00s)
=== RUN   Test_atomicValue_Int
--- PASS: Test_atomicValue_Int (0.00s)
=== RUN   Test_atomicValue_Float
--- PASS: Test_atomicValue_Float (0.00s)
=== RUN   Test_atomicValue_String
--- PASS: Test_atomicValue_String (0.00s)
=== RUN   Test_atomicValue_Duration
--- PASS: Test_atomicValue_Duration (0.00s)
=== RUN   Test_atomicValue_Scan
--- PASS: Test_atomicValue_Scan (0.00s)
PASS
ok  	github.com/go-kratos/kratos/v2/config	0.013s
=== RUN   TestEnvWithPrefix
=== RUN   TestEnvWithPrefix/test_$KEY
=== RUN   TestEnvWithPrefix/test_${KEY:DEFAULT}_without_default
=== RUN   TestEnvWithPrefix/test_${KEY:DEFAULT}_with_default
=== RUN   TestEnvWithPrefix/test_${KEY}_in_array
--- PASS: TestEnvWithPrefix (0.00s)
    --- PASS: TestEnvWithPrefix/test_$KEY (0.00s)
    --- PASS: TestEnvWithPrefix/test_${KEY:DEFAULT}_without_default (0.00s)
    --- PASS: TestEnvWithPrefix/test_${KEY:DEFAULT}_with_default (0.00s)
    --- PASS: TestEnvWithPrefix/test_${KEY}_in_array (0.00s)
=== RUN   TestEnvWithoutPrefix
panic: sync/atomic: store of inconsistently typed value into Value

goroutine 21 [running]:
sync/atomic.(*Value).Store(0xc0002a5040, 0x132c0c0, 0xc00001c2b8)
	/Users/elvizlai/.g/go/src/sync/atomic/value.go:77 +0x110
github.com/go-kratos/kratos/v2/config.(*config).watch.func1(0x132f200, 0xc0002a5050, 0x1374580, 0xc0002a5040, 0x1)
	/Users/elvizlai/temp/kratos/config/config.go:85 +0x12d
sync.(*Map).Range(0xc0002b4048, 0xc00010bf98)
	/Users/elvizlai/.g/go/src/sync/map.go:345 +0x1dc
github.com/go-kratos/kratos/v2/config.(*config).watch(0xc0002b4000, 0x1405b38, 0xc0002a4830)
	/Users/elvizlai/temp/kratos/config/config.go:81 +0x229
created by github.com/go-kratos/kratos/v2/config.(*config).Load
	/Users/elvizlai/temp/kratos/config/config.go:110 +0x136
FAIL	github.com/go-kratos/kratos/v2/config/env	0.016s
=== RUN   TestFile
    file_test.go:168: /var/folders/hs/cwct781947971m14gv8vm8s00000gn/T/test_config/test.json
    file_test.go:168: /var/folders/hs/cwct781947971m14gv8vm8s00000gn/T/test_config
    file_test.go:176: no expected:
        {
            "test":{
                "server":{
        			"name":"$SERVICE_NAME",
                    "addr":"${ADDR:127.0.0.1}",
                    "port":"${PORT:8080}"
                }
            },
            "foo":[
                {
                    "name":"Tom",
                    "age":"${AGE}"
                }
            ]
        }, but got:
        {
            "test":{
                "settings":{
                    "int_key":1000,
                    "float_key":1000.1,
                    "duration_key":10000,
                    "string_key":"string_value"
                },
                "server":{
                    "addr":"127.0.0.1",
                    "port":8000
                }
            },
            "foo":[
                {
                    "name":"nihao",
                    "age":18
                },
                {
                    "name":"nihao",
                    "age":18
                }
            ]
        }
    file_test.go:105: /var/folders/hs/cwct781947971m14gv8vm8s00000gn/T/test_config/test.json
    file_test.go:143: /var/folders/hs/cwct781947971m14gv8vm8s00000gn/T/test_config
    file_test.go:144: /var/folders/hs/cwct781947971m14gv8vm8s00000gn/T/test_config/test.json
--- FAIL: TestFile (0.00s)
=== RUN   TestConfig
    file_test.go:290: {{{1000 1000.1 10000 string_value} {127.0.0.1 8000}} [{nihao 18} {nihao 18}]}
    file_test.go:259: not_found_key not match: key not found
--- PASS: TestConfig (0.00s)
FAIL
FAIL	github.com/go-kratos/kratos/v2/config/file	0.014s
FAIL
```

#### What you expected to happen:
Test case pass.

#### How to reproduce it (as minimally and precisely as possible):
```
cd config
# not every run will trigger FAIL
go test -v -count=1 ./...
```


#### Environment:
- Kratos version (use `kratos -v`): master
- Go version (use `go version`):  go version go1.16.6 darwin/amd64
- OS (e.g: `cat /etc/os-release`): Mac OS X
- Others:

--- a/config/env/env_test.go
+++ b/config/env/env_test.go
@@ -31,7 +31,7 @@ const _testJSON = `
 
 func TestEnvWithPrefix(t *testing.T) {
 	var (
-		path     = filepath.Join(os.TempDir(), "test_config")
+		path     = filepath.Join(t.TempDir(), "test_config")
 		filename = filepath.Join(path, "test.json")
 		data     = []byte(_testJSON)
 	)
@@ -143,7 +143,7 @@ func TestEnvWithPrefix(t *testing.T) {
 
 func TestEnvWithoutPrefix(t *testing.T) {
 	var (
-		path     = filepath.Join(os.TempDir(), "test_config")
+		path     = filepath.Join(t.TempDir(), "test_config")
 		filename = filepath.Join(path, "test.json")
 		data     = []byte(_testJSON)
 	)

--- a/config/file/file_test.go
+++ b/config/file/file_test.go
@@ -84,7 +84,7 @@ const (
 
 func TestFile(t *testing.T) {
 	var (
-		path = filepath.Join(os.TempDir(), "test_config")
+		path = filepath.Join(t.TempDir(), "test_config")
 		file = filepath.Join(path, "test.json")
 		data = []byte(_testJSON)
 	)
@@ -178,7 +178,7 @@ func testSource(t *testing.T, path string, data []byte) {
 }
 
 func TestConfig(t *testing.T) {
-	path := filepath.Join(os.TempDir(), "test_config.json")
+	path := filepath.Join(t.TempDir(), "test_config.json")
 	defer os.Remove(path)
 	if err := ioutil.WriteFile(path, []byte(_testJSON), 0666); err != nil {
 		t.Error(err)
@@ -192,7 +192,7 @@ func TestConfig(t *testing.T) {
 }
 
 func testConfig(t *testing.T, c config.Config) {
-	var expected = map[string]interface{}{
+	expected := map[string]interface{}{
 		"test.settings.int_key":      int64(1000),
 		"test.settings.float_key":    float64(1000.1),
 		"test.settings.string_key":   "string_value",
@@ -258,7 +258,6 @@ func testConfig(t *testing.T, c config.Config) {
 	if _, err := c.Value("not_found_key").Bool(); errors.Is(err, config.ErrNotFound) {
 		t.Logf("not_found_key not match: %v", err)
 	}
-
 }
 
 func testScan(t *testing.T, c config.Config) {

